File: 283moveZeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        j=0
        for i in range(len(nums)):
            if nums[i]!=0:
                nums[j] = nums[i]
                j += 1
        for i in range(j, len(nums)):
            nums[i] = 0



class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        for i in range(nums.count(0)):
            nums.remove(0)
            nums.append(0)
  

File: cpp/283moveZeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int i, j = 0;
        for (i = 0; i < nums.size(); i++) {
            if (nums[i] != 0)
                nums[j++] = nums[i];
        }
        while (j<nums.size())
            nums[j++] = 0;
    }
};

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int i, j = 0;
        for (i = 0; i < nums.size(); i++) {
            if (nums[i]) {
                if (i != j)
                    swap(nums[j++], nums[i]);
                else
                    j++;
            }
        }
    }
};